conda_libmamba_solver/solver.py

```diff
@@ -198,7 +198,11 @@ def solve_final_state(
                     all_channels.append(channel)
         all_channels.extend(in_state.maybe_free_channel())
 
-        all_channels = tuple(all_channels)
+        distinct_channels = OrderedDict()
+        for c in all_channels:
+            if c not in distinct_channels:
+                distinct_channels[c] = True
+        all_channels = tuple(distinct_channels.keys())
```

Comment:
You can dedup while keeping order with:

```suggestion
        all_channels = tuple(dict.fromkeys(all_channels))
```

Reply:
Thanks for the suggestion I didn't know that.  I've gone ahead and changed it to this